File: spring-graphql/src/main/java/com/example/demo/gql/scalars/UUIDScalar.java
package com.example.demo.gql.scalars;

import graphql.language.StringValue;
import graphql.schema.Coercing;
import graphql.schema.CoercingParseLiteralException;
import graphql.schema.CoercingParseValueException;
import graphql.schema.CoercingSerializeException;

import java.util.UUID;

public class UUIDScalar implements Coercing<UUID, String> {
    @Override
    public String serialize(Object o) throws CoercingSerializeException {
        if (o instanceof UUID) {
            return ((UUID) o).toString();
        } else {
            throw new CoercingSerializeException("Not a valid UUID");
        }
    }

    @Override
    public UUID parseValue(Object o) throws CoercingParseValueException {
        return UUID.fromString(o.toString());
    }

    @Override
    public UUID parseLiteral(Object input) throws CoercingParseLiteralException {
        if (input instanceof StringValue) {
            return UUID.fromString(((StringValue) input).getValue());
        }

        throw new CoercingParseLiteralException("Value is not a valid UUID string");
    }
}

File: spring-graphql-rsocket/src/main/kotlin/com/example/demo/gql/scalars/UUIDScalar.kt
package com.example.demo.gql.scalars

import graphql.language.StringValue
import graphql.schema.Coercing
import graphql.schema.CoercingParseLiteralException
import graphql.schema.CoercingSerializeException
import java.util.*

class UUIDScalar : Coercing<UUID, String> {
    override fun serialize(o: Any): String {
        return when (o) {
            is UUID -> o.toString()
            else -> throw CoercingSerializeException("Not a valid UUID")
        }
    }

    override fun parseValue(o: Any): UUID {
        return UUID.fromString(o.toString())
    }

    override fun parseLiteral(input: Any): UUID {
        if (input is StringValue) {
            return UUID.fromString(input.value)
        }
        throw CoercingParseLiteralException("Value is not a valid UUID string")
    }
}